### PR TITLE
add shard compatibility to statSTR runner

### PR DESCRIPTION
**Pull request by hopedisastro**
Add shard compatibility to statSTR to speed it up (creates parallel jobs to mimic multi-threading)

### CODE REVIEW

**Comment by hopedisastro** [comment]
https://batch.hail.populationgenomics.org.au/batches/432221

File: str/trtools/stat_str_runner.py [hunk 1]
```diff
@@ -23,42 +24,65 @@
...
 
-    output_path_vcf = output_path(f'statSTR_samples_{caller}', 'analysis')
-    b.write_output(trtools_job.ofile, output_path_vcf)
+    if sharded:
+        input_vcf_dict = [
+            str(gspath) for gspath in to_path(f'{input_dir}').glob('*.vcf.gz')
+        ]
+
+    else:
+        input_vcf_dict = [input_dir]
```
Comment (michael-harper) [inline str/trtools/stat_str_runner.py:55]: Should it be `[to_path(f'{input_dir}']` here as well?

**Comment by hopedisastro** [comment]
There is a bug for chrY parsing because the genotypes are hemizygous; created an issue [here](https://github.com/gymrek-lab/TRTools/issues/207)

**Review by michael-harper — APPROVED** [review]
Yep looks good to me!

File: str/trtools/stat_str_runner.py (hunk 1 above)
Comment (hopedisastro) [reply str/trtools/stat_str_runner.py:55]: Yes, thanks!

**Pull request merged**